From 5d25b7fd6d53b620b25bed3a72d739d1644b5731 Mon Sep 17 00:00:00 2001
From: Oliver Freyermuth <o.freyermuth@googlemail.com>
Date: Fri, 11 Feb 2022 17:22:13 +0100
Subject: [PATCH] CMake: Fix G4Root_BUILD_TEST option in main CMakeLists.

The option was not correctly inherited for use by the g4root
subdirectory. Fixed by:
- Setting it before add_subdirectory().
- Storing the value in CACHE (otherwise, not seen in subdirectory).

Pull Request: https://github.com/vmc-project/geant4_vmc/pull/28
Bug: https://bugs.gentoo.org/833126
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -60,8 +60,8 @@ include(VMCInstallLibDir)
 #--- Add the packages sources --------------------------------------------------
 #
 if(Geant4VMC_BUILD_G4Root)
+  set(G4Root_BUILD_TEST ${Geant4VMC_BUILD_G4Root_TEST} CACHE BOOL "Build G4Root test")
   add_subdirectory(g4root)
-  set(G4Root_BUILD_TEST ${Geant4VMC_BUILD_G4Root_TEST})
 endif(Geant4VMC_BUILD_G4Root)
 
 if(Geant4VMC_BUILD_Geant4VMC)
-- 
2.34.1

